sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/HttpClient.java

```diff
@@ -20,11 +22,38 @@ public interface HttpClient {
     Response<?> send(HttpRequest request);
 
     /**
-     * Creates a new {@link HttpClient} instance.
+     * Creates an instance of the default {@link HttpClient} implementation, entirely unrelated to all other instances
+     * that were created previously.
      *
-     * @return A new {@link HttpClient} instance.
+     * @return An instance of the default {@link HttpClient} implementation, entirely unrelated to all other instances
+     * that were created previously.
      */
-    static HttpClient createDefault() {
-        return HttpClientProviders.createInstance();
+    static HttpClient getDefault() {
```

Comment:
Might have been part of an offline discussion I wasn't part of, but not a fan of changing the prefix from `create` to `get` as the infer different meanings. This will _always_ create a new instance, so to me it should be called create. `get` infers that you are getting something that is shared, or at least won't be instantiating a new object.

Reply:
I think the intent was to have a shared client and not create a new instance which is why the name was changed from create to get. Is there a reason we want to create new instance for every call?

Reply:
We don't need to create a new client for every call if client sharing is enabled.